fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/ApplyRuleJob.java

```diff
@@ -74,6 +74,11 @@ public final void execute() throws AnalysisException {
         GroupExpressionMatching groupExpressionMatching
                 = new GroupExpressionMatching(rule.getPattern(), groupExpression);
         for (Plan plan : groupExpressionMatching) {
+            if (rule.isExploration()
```

Comment:
if the checking is moved here, can the logic in getJoinRules be removed?

Reply:
no, in OptimizeGroupExpressionJob we could avoid generate ApplyRuleJob at all. so remove it will generate many useless job to slow down planning